feature-appcoins/src/main/java/cm/aptoide/pt/appcoins/presentation/AppCoinsViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package cm.aptoide.pt.appcoins.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import cm.aptoide.pt.appcoins.bonus.BonusAppcModel
+import cm.aptoide.pt.appcoins.bonus.BonusAppcService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AppCoinsViewModel @Inject constructor(
+  private val bonusAppcService: BonusAppcService,
```

Comment:
I would create a use case and not use the service directly

Reply:
Not a big issue here..